pkg/stanza/fileconsumer/config.go

```diff
@@ -80,6 +80,7 @@ type Config struct {
 	FlushPeriod        time.Duration   `mapstructure:"force_flush_period,omitempty"`
 	Header             *HeaderConfig   `mapstructure:"header,omitempty"`
 	DeleteAfterRead    bool            `mapstructure:"delete_after_read,omitempty"`
+	ReplayFile         bool            `mapstructure:"replay_file,omitempty"`
```

Comment:
I'd like to keep this flag out of the user facing API for the package, since it's not clear that it's broadly applicable.

What I'd like to do is put an options pattern in place, which will help with several other changes I anticipate for this package. Give me a day or two to work out what this looks like and then we can rebase to make this an option.

Reply:
I opened https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31596 to establish an options pattern. Once that is merged, let's add a `WithNoTracking` option. (For context on the suggested name, see [#31506](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31506#pullrequestreview-1917375290).) Then we can just pass in this option from only otlpjson receiver, [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/f27143fae73a4a0631db8c55e46f6821d6121f76/receiver/otlpjsonfilereceiver/file.go#L76).

Reply:
I agree with this pattern here, I think this replay option is only useful for the `otlpjsonfilereceiver`. Thanks for being so quick

Reply:
#31596 has been merged

Reply:
d87a14c6ab4c7c80aa41b04c50a1bcb219315995 makes it so that when `noTracking` is set using the option, no known or previous files are tracked. Wasn't sure if we should wait for #31506 to land so the manager doesn't have to make these `!noTracking` checks everywhere. But I'm fine either way

Reply:
Good point. Might be easier overall to wait for #31506

Reply:
Sweet, I'll update this PR today or tomorrow!

Reply:
I lied earlier, i got distracted instead. I did update it now, PTAL